<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Metaphor Computational Art</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>5</number>
    </property>
    <item>
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Timeline</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QScrollArea" name="timeline">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>80</height>
         </size>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="timelineContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>730</width>
           <height>76</height>
          </rect>
         </property>
         <layout class="QHBoxLayout" name="timelineLayout">
          <item>
           <widget class="QToolButton" name="addAxiom">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Add axiom</string>
            </property>
            <property name="statusTip">
             <string>Add axiom</string>
            </property>
            <property name="text">
             <string>Add Axiom</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="axiomList">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Select axiom</string>
            </property>
            <property name="statusTip">
             <string>Select axiom</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QComboBox" name="editAxiom">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip">
             <string>Select axiom</string>
            </property>
            <property name="statusTip">
             <string>Select axiom</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="generateImage">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Generate image</string>
        </property>
        <property name="statusTip">
         <string>Generate image</string>
        </property>
        <property name="text">
         <string>Go!</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Workspace</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QScrollArea" name="workspace">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="workspaceContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>772</width>
         <height>274</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="workspaceLayout">
        <item>
         <widget class="QToolButton" name="newBranch">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>180</width>
            <height>200</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>180</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Add new branch</string>
          </property>
          <property name="statusTip">
           <string>Add new branch</string>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="text">
           <string>Add
New
Branch</string>
          </property>
          <property name="checkable">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Palette</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QScrollArea" name="palette">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>100</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>200</height>
       </size>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="paletteContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>772</width>
         <height>96</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="paletteLayout">
        <item>
         <widget class="QToolButton" name="newProduction">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Add production</string>
          </property>
          <property name="statusTip">
           <string>Add production</string>
          </property>
          <property name="text">
           <string>New Production</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuNew">
     <property name="toolTip">
      <string>New...</string>
     </property>
     <property name="statusTip">
      <string>New...</string>
     </property>
     <property name="title">
      <string>New</string>
     </property>
     <addaction name="actionCurrent_Context"/>
     <addaction name="separator"/>
     <addaction name="actionTurtle"/>
     <addaction name="actionBranch_3D"/>
     <addaction name="actionDla_3D"/>
    </widget>
    <widget class="QMenu" name="menuOpen_Recent">
     <property name="toolTip">
      <string>Open recent...</string>
     </property>
     <property name="statusTip">
      <string>Open recent...</string>
     </property>
     <property name="title">
      <string>Open Recent</string>
     </property>
     <addaction name="separator"/>
     <addaction name="actionClear_Menu"/>
    </widget>
    <addaction name="menuNew"/>
    <addaction name="actionOpen"/>
    <addaction name="menuOpen_Recent"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="actionClose"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuGenerate">
    <property name="title">
     <string>Generate</string>
    </property>
    <addaction name="actionGenerate_Image"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp_Topics"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuGenerate"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="statusTip">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="statusTip">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="toolTip">
    <string>Save As...</string>
   </property>
   <property name="statusTip">
    <string>Save As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="statusTip">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q, Alt+F4</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="statusTip">
    <string>Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionTurtle">
   <property name="text">
    <string>Turtle</string>
   </property>
   <property name="toolTip">
    <string>New from turtle context</string>
   </property>
   <property name="statusTip">
    <string>New from turtle context</string>
   </property>
  </action>
  <action name="actionBranch_3D">
   <property name="text">
    <string>Branch 3D</string>
   </property>
   <property name="toolTip">
    <string>New from branch 3D context</string>
   </property>
   <property name="statusTip">
    <string>New from branch 3D context</string>
   </property>
  </action>
  <action name="actionDla_3D">
   <property name="text">
    <string>Dla 3D</string>
   </property>
   <property name="toolTip">
    <string>New from dla 3D context</string>
   </property>
   <property name="statusTip">
    <string>New from dla 3D context</string>
   </property>
  </action>
  <action name="actionClear_Menu">
   <property name="text">
    <string>Clear Menu</string>
   </property>
   <property name="toolTip">
    <string>Clear recent items menu</string>
   </property>
   <property name="statusTip">
    <string>Clear recent items menu</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="statusTip">
    <string>About</string>
   </property>
  </action>
  <action name="actionHelp_Topics">
   <property name="text">
    <string>Help Topics</string>
   </property>
   <property name="toolTip">
    <string>Help topics</string>
   </property>
   <property name="statusTip">
    <string>Help topics</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionGenerate_Image">
   <property name="text">
    <string>Generate Image</string>
   </property>
   <property name="toolTip">
    <string>Generate image</string>
   </property>
   <property name="statusTip">
    <string>Generate image</string>
   </property>
  </action>
  <action name="actionCurrent_Context">
   <property name="text">
    <string>Current Context</string>
   </property>
   <property name="toolTip">
    <string>New from current context</string>
   </property>
   <property name="statusTip">
    <string>New from current context</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCurrent_Context</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>newFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBranch_3D</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>newBranch3D()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClear_Menu</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>clearMenu()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>closeCurrent()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDla_3D</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>newDla3D()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionGenerate_Image</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>generateImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionHelp_Topics</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>help()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>openFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveCurrentFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveFileAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTurtle</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>newTurtle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newBranch</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addBranch()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>211</y>
    </hint>
    <hint type="destinationlabel">
     <x>263</x>
     <y>122</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newProduction</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addProduction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>66</x>
     <y>531</y>
    </hint>
    <hint type="destinationlabel">
     <x>158</x>
     <y>466</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generateImage</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>generateImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>762</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>698</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addAxiom</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addAxiom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>155</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>axiomList</sender>
   <signal>activated(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>selectAxiom(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>154</x>
     <y>98</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editAxiom</sender>
   <signal>activated(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>insertAxiom(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>680</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>newFile()</slot>
  <slot>openFile()</slot>
  <slot>saveFileAs()</slot>
  <slot>saveCurrentFile()</slot>
  <slot>generateImage()</slot>
  <slot>help()</slot>
  <slot>closeCurrent()</slot>
  <slot>newTurtle()</slot>
  <slot>newBranch3D()</slot>
  <slot>newDla3D()</slot>
  <slot>clearMenu()</slot>
  <slot>about()</slot>
  <slot>addBranch()</slot>
  <slot>addProduction()</slot>
  <slot>addAxiom()</slot>
  <slot>selectAxiom(int)</slot>
  <slot>insertAxiom(int)</slot>
 </slots>
</ui>
